README.md

```diff
@@ -1,5 +1,40 @@
-# lighthouse 
-Controller to facilitate DNS discovery in multi-cluster connected environments (proof-of-concept state)
+# Lighthouse 
+Lighthouse provides DNS discovery to clusters connected by [Submariner](https://github.com/submariner-io/submariner) in multi-cluster connected environments. The solution aims to compatible with any CNI-compatible cluster network provider.
+
+# Architecture
+The below digram shows the basic Lighthouse architecture.
+
+![Lighthouse Architecture](https://raw.githubusercontent.com/submariner-io/lighthouse/master/docs/img/architecture.png)
+
+## Lighthouse Controller
+This is the central discovery controller that gathers the information from the clusters, decides what information is to be shared and distributes the information as newly defined CRDs.
+
+The Lighthouse controller will be running along with the [Admiral](https://github.com/submariner-io/admiral) control plane. It uses the API’s exposed by the Admiral to connect to each cluster and listen for Service CRUD. Once it is notified that a new Service is created and if the Service needs to be distributed, it frames the Lighthouse CRD and the same is distributed to all the clusters that are part of the control plane. The flow for update and delete will be similar.
```

Comment:
- "Once it is notified that a new Service is created and needs to be distributed"
- "it frames the Lighthouse CRD" - What does it mean "frame" ?

Reply:
Done, Changed "frames" to "creates".